### DISCUSSION TITLE
Draw command to create an image from text, but text is cut-off

### DISCUSSION THREAD

**Discussion by Rapster-UK** [ImageMagick/ImageMagick#7695] [Help] [answered]
### ImageMagick version

7.1.1-38

### Operating system, version and so on

Windows 11

### Description

Hi, I'm using the following command to create images containing text on a transparent background, however when the text is over a certain length, it gets truncated. Is it possible to use something like a 'wrap' option to make it wrap onto a new line?

magick -size 1000x1000 canvas:none -font player.ttf -pointsize 72 -quality 100 -gravity center -draw "text 25,60 'This is a test and it is to long and is cut-off!'" -channel RGBA -blur 0x6 -tile "tile-1.jpg" -stroke white -draw "text 20,55 'This is a test and it is to long and is cut-off!'" -trim +repage "Output.png"

Ok:
![Output 1](https://github.com/user-attachments/assets/b92eac16-0148-40a7-99be-d16bcaf51693)

To long:
![Output 2](https://github.com/user-attachments/assets/2e9b2472-a2fe-4bf5-827d-ad01e128bf70)

The draw command is used twice, once to create the main text and again to create the shadow.

**Comment by fmw42** [comment]
Use caption:"your text" to automatically wrap.  -draw and -annotate do not automatically wrap.

See https://imagemagick.org/Usage/text/

**Comment by Rapster-UK** [comment]
Hi, thanks for the info, I've updated the command and it now wraps the text, which is great. However, the -tile command no seems to take affect, is it supported with caption? If not, is there another way to fill the text with a pattern? I can use -fill but that's only solid colours.

New command:
magick -size 1000x1000 -background none -font player.ttf -pointsize 72 -quality 100 -gravity center caption:"This is a test and it is to long and is cut-off!" -channel RGBA -blur 0x6 -tile "tile-1.jpg" -stroke white caption:"This is a test and it is to long and is cut-off!" -flatten -trim +repage "Output.png"

![Output](https://github.com/user-attachments/assets/be4db7b1-9e8a-4163-971a-120a7b38e30f)

**Comment by snibgo** [comment]
Please post `tile-1.jpg`.

**Comment by Rapster-UK** [comment]
Here you go :)

![tile-1](https://github.com/user-attachments/assets/d226eba2-6b65-48c4-825b-c252f755d654)

**Comment by Rapster-UK** [comment]
and the font used...

[player.zip](https://github.com/user-attachments/files/17402315/player.zip)

**Comment by snibgo** [comment, marked as answer]
I suggest this (Windows BAT syntax). It draws the text three times: to get the blurred shadow, to get the coloured letters, and to get the white stroked letters.
```
magick ^
  -font player.ttf -pointsize 72 ^
  -size 1000x1000 ^
  -gravity center ^
  -background none ^
  "caption:This is a test and it is too long and is cut-off^!" ^
  -channel RGBA -blur 0x6 ^
  ( tile:tile-1.jpg ^
    -background Black ^
    -stroke White -fill White ^
    "caption:This is a test and it is too long and is cut-off^!" ^
    -alpha off -compose CopyOpacity -composite ^
  ) ^
  -compose Over -composite ^
  -background None ^
  -stroke White -fill None ^
  "caption:This is a test and it is too long and is cut-off^!" ^
  -compose Over -composite ^
  -trim +repage ^
  x.png
```
The result x.png is:

![x](https://github.com/user-attachments/assets/6633aae8-b3af-4c45-a4f1-d826a5521323)

**Reply by Rapster-UK** [reply]
That's perfect! Thank you!!!
